app/javascript/single-file-upload.js

```diff
@@ -0,0 +1,101 @@
+window.addEventListener("DOMContentLoaded", (event) => {
+  const fileInput = document.querySelector('.govuk-file-upload');
+  const maxFileSize = fileInput.dataset.maxSize;
+  const errorHeading = fileInput.dataset.errorHeading;
+  const errorMessage = fileInput.dataset.sizeErrorMessage;
+  const feedback = $(".single-file-upload__message");
+  const saveButtons = document.querySelectorAll('button[type="submit"]');
+
+  if(fileInput){
+    /*
+      logic for file progress indicator
+      and uploaded files list
+    */
+    fileInput.addEventListener('change', function(event) {
+        if(fileInput.files.length > 0){
+          handleUploadList(fileInput);
+        }
+    });
+
+    /*
+      handle checking file size, if exceeds limit,
+      refresh with error shown and move page to file upload section.
+    */
+    if (saveButtons) {
+      for(const button of saveButtons) {
+        button.addEventListener('click', function (event) {
+          if(fileInput.files.length > 0){
+            if(!validateFileSize(fileInput, maxFileSize, feedback)){
+              generateFileSizeError(fileInput, errorMessage, errorHeading, feedback);
+              event.preventDefault();
+              event.stopImmediatePropagation();
+            }
+          }
+        })
+      }
+    }
+  }
+});
+
+function handleUploadList(fileInput){
+  //replace selected file text
+  let filenameCell = document.querySelector('.moj-multi-file-upload__filename');
+  let uploadedFile = fileInput.files[0];
+  if(uploadedFile){
+    filenameCell.textContent = uploadedFile.name;
+  }
+}
+
+function validateFileSize(fileInput, maxFileSize, feedback){
+  if (fileInput) {
+    feedback.html('');
+    removeInlineError(fileInput);
+    const file = fileInput.files[0];
+    let validationPassed = file && (file.size <= maxFileSize)
+
+    return validationPassed;
+  }
+}
+
+function generateFileSizeError(fileInput, errorMessage, errorHeading, feedback) {
+  feedback.html(govukErrorSummary(fileInput.id, errorMessage, errorHeading));
+  addInlineError(fileInput, errorMessage)
+  let errorSummary = document.querySelector('.govuk-error-summary');
+  errorSummary.scrollIntoView();
```

Comment:
`errorSummary.scrollIntoView({ behavior: 'smooth' });` will be less visually jarring. Also no null check here.